bin/wabur

```diff
@@ -36,7 +42,7 @@ options = {
   },
   store: {
     dir: {
-      val: File.join('$BASE', 'data'),
+      val: '$BASE/data',
```

Comment:
Hardcoding the file path separator will break on Windows unless Ruby now does the translation under the covers.

Reply:
This string is only going to be used to display the option in the `--help` command..
The actual value is going to be within a config file or passed via the `--http.dir`... so the `File.join` should actually be where it belongs.. in the `Shell#initialize`

Reply:
That value is also used as the default so it will be used if the user does not over-ride it.

Reply:
`File.join('$BASE', 'data')` is going to return `$BASE/data` no matter what.. if we really need the `$BASE` to evaluated, then it should actually be..
`File.join($BASE, 'data')`.. but then `$BASE` is just an arbitrary global..

Reply:
Base is just a string and the sub called later replaces it with the value of base. The join should use the appropriate file path separator for the system the code is running on so for Windows it would use a backslash.

Reply:
My bad.. I didn't see the parentheses in the code below:
```ruby
@http_dir     = (config['http.dir'] || File.join(base, 'pages')).gsub('$BASE', base)
```

Reply:
now the question is what happens if a Windows user passes the following via CLI.. or has similar in a config file:?
```
wabur --http.dir foo/bar --base lorem
```

Reply:
I would assume a Windows user would use Windows file paths and nor Unix file paths.

Reply:
assumption is a bad thing..
one cant blame a user for sending in UNIX style paths in a UNIX style `.conf` file..

Reply:
You are assuming Windows users will use Unix file paths. I think that assumption is less likely to be true. Now, back to the change. As the default value, it will fail on Windows. Why would we want that?

Reply:
I don't think it'll break on Windows.. because `File::SEPARATOR = "/"` even on Windows.. 
searching for documentary evidence..

besides, I still think `File.join` is unnecessary here..
We might as well handle it robustly in Shell class...

Reply:
The shell class should not have to second guess what was meant when given a file path. It should be correct as given in the configuration or else it should error out. 

You have a Windows machine there. What do you get when you File.join and can the result be opened?

Reply:
the following is in an irb session in `cmd.exe` on a Windows machine: (not even Git BASH)
```
irb(main):001:0>
irb(main):002:0*
irb(main):003:0*
irb(main):004:0*
irb(main):005:0*
irb(main):006:0*
irb(main):007:0* string1 = 'foo'
=> "foo"
irb(main):008:0> string2 = 'bar'
=> "bar"
irb(main):009:0> File.join(string1, string2)
=> "foo/bar"
irb(main):010:0>
```

Reply:
An can that be opened on Windows? If so why ever use File.join any more? In the old days File.join was needed to get a path that opened on Windows.

Reply:
I synced my local branch with the current state here, ran `script/start-sample`, opened `http://localhost:6363` and successfully added a record to `Article` list..

Reply:
Good enough. No more use of File.join then.

Reply:
IMO, if we really want to avoid any unforeseen breakages surrounding this, something similar perhaps:
```ruby
File.join(config['http.dir'].split(/\\|\//))
```
..but just looking at the above code makes my eyes burn.. :laughing:

Reply:
Please no. We know / is the separator on all platforms so don't bother with File.join. I think we are done with this.